### DISCUSSION TITLE
Averaging of Measurements

### DISCUSSION THREAD

**Discussion by sanschaussure** [pvvx/ATC_MiThermometer#460] [Q&A]
Good morning, 

I just received 4 LYWSD03MMC. I flashed them with version 4.6 and everything works smoothly. Now I tested them and see a really slow reaction to changes of temperature and humidity. I change the position of the devices from inside to outside of the house (~ delta T 10°C) and the outside value is first reached after ~30 minutes (see attachment). 

Is there any rolling average in the firmware or is the delay only because of the temperature adaption of the whole device? If this is an firmware feature, is there any option to change the period of the averaging? I try changing the averaging option in the setting but I see no influence to the sent values. 

Thanks for your answers.

![Bildschirmfoto_2024-01-04_12-08-07](https://github.com/pvvx/ATC_MiThermometer/assets/46602102/5ee50d48-78cf-4e87-af90-8178d8361a54)

**Comment by lbarjak** [comment]
Hello, this is thermal inertia.

sanschaussure ***@***.***> ezt írta (időpont: 2024. jan. 4.,
Cs, 12:11):

> Good morning,
>
> I just received 4 LYWSD03MMC. I flashed them with version 4.6 and
> everything works smoothly. Now I tested them and see a really slow reaction
> to changes of temperature and humidity. I change the position of the
> devices from inside to outside of the house (~ delta T 10°C) and the
> outside value is first reached after ~30 minutes (see attachment).
>
> Is there any rolling average in the firmware or is the delay only because
> of the temperature adaption of the whole device? If this is an firmware
> feature, is there any option to change the period of the averaging? I try
> changing the averaging option in the setting but I see no influence to the
> sent values.
>
> Thanks for your answers.
>
> Bildschirmfoto_2024-01-04_12-08-07.png (view on web)
> <https://github.com/pvvx/ATC_MiThermometer/assets/46602102/5ee50d48-78cf-4e87-af90-8178d8361a54>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pvvx/ATC_MiThermometer/discussions/460>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE3B7YOFSDM3J2LX2F4CRHTYM2E6VAVCNFSM6AAAAABBMYLH6OVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGAZDOOJYG4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Barják László   +36209347499

**Comment by sanschaussure** [comment]
Thank your for the quick response. Is it only thermal inertia? The heat capacity and mass of the plastic case is in my opinion not big enough for and half hour asymptotic slope. 

Regards.

**Comment by pvvx** [comment]
The SHTv3 sensor is faster than the SHT4x.
The heat capacity is formed by CR2032.

**Comment by Lassska** [comment]
you will have to choose a very small hysteresis, for example 0.1°C, to use an MI2 as a thermostat, otherwise the heating cycles will last around 20 minutes and the "true" peak (blue), measured with a digital sensor without the case will be far beyond the expected comfort. ..
![Screenshot_20240311_205359_Chrome](https://github.com/pvvx/ATC_MiThermometer/assets/61229417/2c365315-cb54-4cc7-bd11-227db2783e52)

**Comment by pvvx** [comment]
20C in the pantry:
![image](https://github.com/pvvx/ATC_MiThermometer/assets/12629515/aeef134a-1c06-4f50-97b2-a6b9681728a2)

The beginning of the graph is working with the sensor: ![image](https://github.com/pvvx/ATC_MiThermometer/assets/12629515/2264d63a-110b-4688-9dc6-f9457e858207)
Next - in HA with Xiaomi LYWSD03MMC.
The thermostat in HA has a minimum hysteresis of 0.1C. This is not enough, since 0.01..0.02 C is needed.
The inertia of the heater has a particularly strong effect.

![image](https://github.com/pvvx/ATC_MiThermometer/assets/12629515/22c5fce6-6e14-4d76-a00c-babbc6d34f56)

**Reply by Lassska** [reply]
:)
![Screenshot_20240312_100522_Chrome](https://github.com/pvvx/ATC_MiThermometer/assets/61229417/57815b27-3b9e-409d-b932-ab134f047572)
![20240312_082915](https://github.com/pvvx/ATC_MiThermometer/assets/61229417/ca9251aa-cdbd-41ff-a7dd-a6a18bf62a32)
